packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -49,25 +78,129 @@ export function Comments( {
 	showCommentBoard,
 	setShowCommentBoard,
 } ) {
-	const { blockCommentId } = useSelect( ( select ) => {
-		const { getBlockAttributes, getSelectedBlockClientId } =
-			select( blockEditorStore );
-		const _clientId = getSelectedBlockClientId();
+	const { blockCommentId, selectedBlockClientId, blocks } = useSelect(
+		( select ) => {
+			const { getBlockAttributes, getSelectedBlockClientId, getBlocks } =
+				select( blockEditorStore );
+			const _clientId = getSelectedBlockClientId();
+			return {
+				blockCommentId: _clientId
+					? getBlockAttributes( _clientId )?.blockCommentId
+					: null,
+				selectedBlockClientId: _clientId,
+				blocks: getBlocks(),
+			};
+		},
+		[]
+	);
 
-		return {
-			blockCommentId: _clientId
-				? getBlockAttributes( _clientId )?.blockCommentId
-				: null,
-		};
-	}, [] );
+	const { toggleBlockHighlight, flashBlock } =
+		useDispatch( blockEditorStore );
+
+	const clearThreadFocus = () => {
+		// Clear any block highlights when clearing thread focus
+		if ( focusThread && blocks ) {
+			const relatedBlockClientId = findBlockByCommentId(
+				focusThread,
+				blocks
+			);
+			if ( relatedBlockClientId ) {
+				toggleBlockHighlight( relatedBlockClientId, false );
+			}
+		}
+		setFocusThread( null );
+		setShowCommentBoard( false );
+	};
 
 	const [ focusThread, setFocusThread ] = useState(
 		showCommentBoard && blockCommentId ? blockCommentId : null
 	);
 
-	const clearThreadFocus = () => {
-		setFocusThread( null );
+	// Sync comment ↔ block highlighting bidirectionally.
+	useEffect( () => {
+		const relatedBlockClientId =
+			focusThread && blocks
+				? findBlockByCommentId( focusThread, blocks )
+				: null;
+
+		// Clear highlight if switching blocks.
+		if (
+			relatedBlockClientId &&
+			relatedBlockClientId !== selectedBlockClientId
+		) {
+			toggleBlockHighlight( relatedBlockClientId, false );
```

Comment:
```suggestion
				: null;
```

Now we're using `flashBlock`, so I think we don't need  `toggleBlockHighlight`.